src/tests/test_ecies.cpp

```diff
@@ -0,0 +1,246 @@
+      result.test_eq("decrypted data equals plaintext", decrypted, plaintext);
+      }
+   catch(Botan::Lookup_Error& e)
+      {
+      result.test_note(std::string("Test not executed: ") + e.what());
+      }
+   }
+
+void check_encrypt_decrypt(Test::Result& result, const Botan::ECDH_PrivateKey& private_key,
+                           const Botan::ECDH_PrivateKey& other_private_key,
+                           const Botan::ECIES_System_Params& ecies_params, size_t iv_length = 0)
+   {
+   static std::vector<byte> Plaintext { 1, 2, 3 };
+   check_encrypt_decrypt(result, private_key, other_private_key, ecies_params, std::vector<byte>(iv_length, 0), "",
+                         Plaintext, std::vector<byte>());
+   }
+
+class ECIES_ISO_Tests : public Text_Based_Test
+   {
+   public:
+      ECIES_ISO_Tests() : Text_Based_Test(
+            "pubkey/ecies-18033.vec",
+         { "format", "p", "a", "b", "mu", "nu", "gx", "gy", "hx", "hy", "x", "r", "C0", "K" })
+         {
+         }
+
+      Test::Result run_one_test(const std::string&, const VarMap& vars) override
+         {
+         Test::Result result("ECIES-ISO");
+
+         // get test vectors defined by ISO 18033
+         const Botan::PointGFp::Compression_Type compression_type = get_compression_type(get_req_str(vars, "format"));
+         const Botan::BigInt p = get_req_bn(vars, "p");
+         const Botan::BigInt a = get_req_bn(vars, "a");
+         const Botan::BigInt b = get_req_bn(vars, "b");
+         const Botan::BigInt mu = get_req_bn(vars, "mu");	// order
+         const Botan::BigInt nu = get_req_bn(vars, "nu");	// cofactor
+         const Botan::BigInt gx = get_req_bn(vars, "gx");	// base point x
+         const Botan::BigInt gy = get_req_bn(vars, "gy");	// base point y
+         const Botan::BigInt hx = get_req_bn(vars, "hx");	// x of public point of bob
+         const Botan::BigInt hy = get_req_bn(vars, "hy");	// y of public point of bob
+         const Botan::BigInt x = get_req_bn(vars, "x");	// private key of bob
+         const Botan::BigInt r = get_req_bn(vars, "r");	// (ephemeral) private key of alice
+         const std::vector<byte> c0 = get_req_bin(vars, "C0");	// expected encoded (ephemeral) public key
+         const std::vector<byte> k = get_req_bin(vars, "K");	// expected derived secret
+
+         const Botan::CurveGFp curve(p, a, b);
+         const Botan::EC_Group domain(curve, Botan::PointGFp(curve, gx, gy), mu, nu);
+
+         // keys of bob
+         const Botan::ECDH_PrivateKey other_private_key(Test::rng(), domain, x);
+         const Botan::PointGFp other_public_key_point(curve, hx, hy);
+         const Botan::ECDH_PublicKey other_public_key(domain, other_public_key_point);
+
+         // (ephemeral) keys of alice
+         const Botan::ECDH_PrivateKey eph_private_key(Test::rng(), domain, r);
+         const Botan::PointGFp eph_public_key_point = eph_private_key.public_point();
+         const std::vector<byte> eph_public_key_bin = Botan::unlock(
```

Comment:
This line certainly doesn't look right - are these tests being included in the CI build?